BluetoothLEExplorer/BluetoothLEExplorer/Models/ObservableBluetoothLEDevice.cs

```diff
@@ -635,13 +648,14 @@ private async Task<bool> GetAllPrimaryServices(BluetoothCacheMode cacheMode)
                         var temp = new ObservableGattDeviceService(serv);
                         // This isn't awaited so that the user can disconnect while the services are still being enumerated
                         temp.Initialize();
-                        returnedServices.Add(temp);
+                        Services.Add(temp);

```

Comment:
>Services [](http://example.com/codeflow?start=24&length=8)

We used to lock `Services` before updating it. Can this be (safely) added-to in this loop without locking? Or do we need to accumulate a new collection of services in this loop and add them to `Services` at the end, under lock?

Reply:
I mean it could, but the lock didn't really do anything because only 2 places mess with Services (Dispose and here), and dispose didn't check the lock. I guess I can go add the lock outside the loop and to Dispose. (I'll go do that.)